streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -72,72 +72,90 @@
 
     public static class Config extends AbstractConfig {
 
-        public static Config fromStreamsConfig(StreamsConfig streamsConfig) {
+        static Config fromStreamsConfig(StreamsConfig streamsConfig) {
             return new Config(streamsConfig.originals());
         }
 
-        public Config(Map<?, ?> originals) {
+        Config(Map<?, ?> originals) {
             super(CONFIG, originals, false);
         }
-    }
 
+    }
     private final KafkaClient kafkaClient;
     private final List<MetricsReporter> reporters;
     private final Config streamsConfig;
-
+    private final Map<String, String> defaultTopicConfigs = new HashMap<>();
     private static final int MAX_INFLIGHT_REQUESTS = 100;
 
-    public StreamsKafkaClient(final StreamsConfig streamsConfig) {
-        this(Config.fromStreamsConfig(streamsConfig));
-    }
 
-    public StreamsKafkaClient(final Config streamsConfig) {
+    StreamsKafkaClient(final Config streamsConfig,
+                       final KafkaClient kafkaClient,
+                       final List<MetricsReporter> reporters) {
         this.streamsConfig = streamsConfig;
+        this.kafkaClient = kafkaClient;
+        this.reporters = reporters;
+        extractDefaultTopicConfigs(streamsConfig.originalsWithPrefix(StreamsConfig.TOPIC_PREFIX));
+    }
 
+    private void extractDefaultTopicConfigs(final Map<String, Object> configs) {
+        for (Map.Entry<String, Object> entry : configs.entrySet()) {
+            if (entry.getValue() != null) {
```

Comment:
I cannot follow here. The given `config` is the complete `StreamsConfig`. So why do we copy all not-null entries? Should we not just copy if `key.startWith(TOPIC_PREFIX)` -- guess I am missing something here.